src/main/java/org/openstreetmap/atlas/checks/database/DatabaseConnection.java

```diff
@@ -0,0 +1,92 @@
+package org.openstreetmap.atlas.checks.database;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+
+/**
+ * Connect and create schema for CheckFlag database
+ *
+ * @author danielbaah
+ */
+public class DatabaseConnection
+{
+
+    private Connection connection;
+    private static final Logger logger = LoggerFactory.getLogger(DatabaseConnection.class);
+
+    /**
+     * Default constructor takes in a url of the form host[:port]/database. Port and additional
+     * query params are optional; i.e, localhost/database would default to port 5432. Also,
+     * additional query parameters can be appended to the database name; i.e.
+     * {@code localhost/database?user=postgres&currentSchema=public&password=private}
+     * 
+     * @see <a href="https://jdbc.postgresql.org/documentation/head/connect.html">JDBC connection
+     *      parameters</a>
+     * @param connectionUrl
+     *            a database url
+     */
+    public DatabaseConnection(final String connectionUrl)
+    {
+        try
+        {
+            final URI connectionURI = this.createConnectionURI(connectionUrl);
+
+            this.connection = DriverManager
+                    .getConnection(String.format("jdbc:%s", connectionURI.toString()));
+            this.createDatabaseSchema();
+        }
+        catch (final SQLException error)
+        {
+            throw new CoreException("Invalid connection string. host[:port]/database", error);
+        }
+    }
+
+    public Connection getConnection()
+    {
+        return this.connection;
+    }
+
+    private URI createConnectionURI(final String connectionString)
+    {
+        return URI.create(String.format("postgresql://%s", connectionString));
+    }
+
+    private boolean createDatabaseSchema()
```

Comment:
Looks like this method can be void instead of boolean. Is the boolean used anywhere?

Reply:
yup, good catch